eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -137,8 +137,8 @@ public function evaluate(APILimitationValue $value, APIUserReference $currentUse
         $objectStateHandler = $this->persistence->objectStateHandler();
         $stateGroups = $objectStateHandler->loadAllGroups();
 
-        // First deal with unpublished content
-        if ($object instanceof ContentCreateStruct || !$object->published) {
+        // First deal with new content
+        if ($object instanceof ContentCreateStruct) {
```

Comment:
Should we also add a comment here that ATM there is no known use-case where `$object` might ever be a `ContentCreateStruct` ? ( There is no object state limitation for `content/create`, only for `content/edit`

Reply:
```suggestion
        // ATM, it seems unclear if $object can ever be a ContentCreateStruct, there are no object state limitation
        // for content/create, only for content/edit.
        // Keeping the block here as there are still Unit tests that inject ContentCreateStruct values to this function
        if ($object instanceof ContentCreateStruct) {
```

Reply:
But by keeping the block, we won't breake BC, only fix the broken behaviour

Reply:
> @alongosz : Not sure if you want such comment in the code, but it might be of help later

Definitely useful.